src/shared/components/query/GeneSetSelector.tsx

```diff
@@ -40,16 +41,29 @@ export default class GeneSetSelector extends QueryStoreComponent<GeneSetSelector
 
 	@computed get geneListOptions()
 	{
-		return [
-			{
-				label: 'User-defined List',
-				value: ''
-			},
-			...gene_lists.map(item => ({
-				label: `${item.id} (${item.genes.length} genes)`,
-				value: item.genes.join(' ')
-			}))
-		];
+	    if (AppConfig.querySetsOfGenes == null) {
+	        return [
+                    {
+                        label: 'User-defined List',
+                        value: ''
+                    },
+                    ...gene_lists.map(item => ({
+                        label: `${item.id} (${item.genes.length} genes)`,
+                        value: item.genes.join(' ')
+                    }))
+                ];
+	    } else {
+	        return [
+	                {
+	                    label: 'User-defined List',
+	                    value: ''
+	                },
+	                ...AppConfig.querySetsOfGenes == undefined ? [{label:"", value:""}] : AppConfig.querySetsOfGenes.map(item => ({
```

Comment:
curious why we want [{label:"", value:""}]  if querySetsofGenes is undefined.

Reply:
This is an error, the first if should contain `null` and `undefined`.